src/sentry/search/snuba/backend.py

```diff
@@ -534,25 +534,37 @@ def _get_queryset_conditions(
             queryset_conditions.update(
                 {"issue.type": QCallbackCondition(lambda types: Q(type__in=types))}
             )
-            queryset_conditions.update(
-                {
-                    "message": QCallbackCondition(
-                        lambda query: Q(type=GroupType.ERROR.value)
-                        | Q(
-                            type__in=(
-                                GroupType.PERFORMANCE_N_PLUS_ONE.value,
-                                GroupType.PERFORMANCE_SLOW_SPAN.value,
-                                GroupType.PERFORMANCE_SEQUENTIAL_SLOW_SPANS.value,
-                                GroupType.PERFORMANCE_LONG_TASK_SPANS.value,
-                                GroupType.PERFORMANCE_RENDER_BLOCKING_ASSET_SPAN.value,
-                                GroupType.PERFORMANCE_DUPLICATE_SPANS.value,
-                                GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value,
-                            ),
-                            message__icontains=query,
-                        )
-                    )
-                }
+            message_filter = next(
+                (sf for sf in search_filters or () if "message" == sf.key.name), None
             )
+            if message_filter:
+
+                def _perf_issue_message_condition(query: str) -> Q:
+                    return Q(
+                        type__in=(
+                            GroupType.PERFORMANCE_N_PLUS_ONE.value,
+                            GroupType.PERFORMANCE_SLOW_SPAN.value,
+                            GroupType.PERFORMANCE_SEQUENTIAL_SLOW_SPANS.value,
+                            GroupType.PERFORMANCE_LONG_TASK_SPANS.value,
+                            GroupType.PERFORMANCE_RENDER_BLOCKING_ASSET_SPAN.value,
+                            GroupType.PERFORMANCE_DUPLICATE_SPANS.value,
+                            GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value,
```

Comment:
can we turn this into an iterator? would be more future proof.

something like:
```
performance_types = []
for group_type, group_category in GROUP_TYPE_TO_CATEGORY:
    if group_category == GroupCategory.PERFORMANCE:
        performance_types.append(group_type.value)

type__in=(performance_types)

Reply:
actually maybe just stick a helper method in `types/issues.py` that takes in a `group category` and returns a list of `group types`

Reply:
Good idea. Will make that change.